lib/daru/index.rb

```diff
@@ -419,4 +501,109 @@ def nils_counted array
       }
     end
   end
-end
+
+  class CategoricalIndex < Index
+    
+    include Enumerable
```

Comment:
BTW, why is it here? Either each index should be enumerable (with thorough check if it is sane at all), or neither. I'm not sure that one specific type of index should include `Enumerable` on its own will.